gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/FieldSchema.java

```diff
@@ -0,0 +1,287 @@
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Google Bigquery schema for a Table field. Several data types and modes are supported.
+ */
+public class FieldSchema implements Serializable {
+
+  static final Function<TableFieldSchema, FieldSchema> FROM_PB_FUNCTION =
+      new Function<TableFieldSchema, FieldSchema>() {
+        @Override
+        public FieldSchema apply(TableFieldSchema pb) {
+          return FieldSchema.fromPb(pb);
+        }
+      };
+  static final Function<FieldSchema, TableFieldSchema> TO_PB_FUNCTION =
+      new Function<FieldSchema, TableFieldSchema>() {
+        @Override
+        public TableFieldSchema apply(FieldSchema fieldSchema) {
+          return fieldSchema.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -8154262932305199256L;
+
+  /**
+   * Data Types for a BigQuery Table field.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatypes">
+   *     Data Types</a>
+   */
+  public enum Type {
+    STRING, INTEGER, FLOAT, BOOLEAN, TIMESTAMP, RECORD
+  }
+
+  /**
+   * Mode for a BigQuery Table field. {@link Mode#NULLABLE} fields can be set to {@code null},
+   * {@link Mode#REQUIRED} fields must be provided. {@link Mode#REPEATED} fields can contain more
+   * than one value.
+   */
+  public enum Mode {
+    NULLABLE, REQUIRED, REPEATED
+  }
+
+  private final String name;
+  private final Type type;
+  private final Mode mode;
+  private final String description;
+  private final List<FieldSchema> fields;
+
+  public static final class Builder {
+
+    private String name;
+    private Type type;
+    private Mode mode;
+    private String description;
+    private List<FieldSchema> fields;
+
+    private Builder() {}
+
+    /**
+     * Sets the field name. The name must contain only letters (a-z, A-Z), numbers (0-9), or
+     * underscores (_), and must start with a letter or underscore. The maximum length is 128
+     * characters.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets the type of the field.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatypes">
+     *     Data Types</a>
+     */
+    public Builder type(Type type) {
+      this.type = checkNotNull(type);
+      return this;
+    }
+
+    /**
+     * Sets the mode of the field. By default {@link Mode#NULLABLE} is used.
+     */
+    public Builder mode(Mode mode) {
+      this.mode = mode;
```

Comment:
Honestly, I think if we are going to use patch for updating a [`Table`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/bigquery/gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/BigQueryRpc.java#L153) we should probably
not relax the implementation based on current restriction as that might get relaxed in the future
and break the code.

If you agree, we don't have to do it now but rather create an issue for it.